java-checks/src/main/java/org/sonar/java/checks/KnownCapacityHashMapsCheck.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+@Rule(key = "S6485")
+public class KnownCapacityHashMapsCheck extends IssuableSubscriptionVisitor {
+
+  private static final MethodMatchers MATCHERS = MethodMatchers.create()
+    .ofTypes("java.util.HashMap", "java.util.HashSet", "java.util.LinkedHashMap", "java.util.LinkedHashSet", "java.util.WeakHashMap")
+    .constructor()
+    .addParametersMatcher("int")
+    .build();
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.NEW_CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+
+    if (!context.getJavaVersion().isJava19Compatible()) {
+      return;
+    }
+
+    NewClassTree newClassTree = (NewClassTree) tree;
+    if (MATCHERS.matches(newClassTree)) {
+      reportIssue(newClassTree, "Replace this call to the constructor with the better suited static method.");
```

Comment:
Let's try and add a more specific issue message to set developers on the right path to a fix